imblearn/base.py

```diff
@@ -231,8 +227,119 @@ def __setstate__(self, dict):
         self.__dict__.update(dict)
         self.logger = logger
 
-    @classmethod
-    def get_properties(cls):
-        """Get the properties for this estimator."""
 
-        return cls._estimator_prop
+class BaseBinaryclassSampler(six.with_metaclass(ABCMeta, SamplerMixin)):
+    """Base class for all binary class sampler.
+
+    Warning: This class should not be used directly. Use derived classes
+    instead.
+
+    """
+
+    def fit(self, X, y):
+        """Find the classes statistics before to perform sampling.
+
+        Parameters
+        ----------
+        X : ndarray, shape (n_samples, n_features)
+            Matrix containing the data which have to be sampled.
+
+        y : ndarray, shape (n_samples, )
+            Corresponding label for each sample in X.
+
+        Returns
+        -------
+        self : object,
+            Return self.
+
+        """
+
+        super(BaseBinaryclassSampler, self).fit(X, y)
+
+        # Check that the target type is binary
+        if not type_of_target(y) == 'binary':
+            warnings.warn('The target type should be binary.')
+
+        return self
+
+
+    @abstractmethod
+    def _sample(self, X, y):
```

Comment:
Nope. The class that will be instantiated should implement it.
You can test it locally.